src/source.cpp

```diff
@@ -46,11 +46,16 @@ namespace metricq
 
 Source::Source(const std::string& token) : DataClient(token)
 {
+    register_management_callback("config", [this](const json& config) -> json {
+        on_source_config(config);
+        declare_metrics();
```

Comment:
At least in the of the lmgd, this `declare_metrics()` call is too early. The list of metrics isn't updated then. This is also reflected in the delayed `declare_metrics()` after `on_source_ready()` during the inital connect.

Reply:
Instead of fixing this in the interface, we decided to fix the manager instead. I, therefore, assume that as resolved.